src/vimscript/pattern.ts

```diff
@@ -83,8 +84,16 @@ export class Pattern {
       fromPosition = args.fromPosition;
     }
 
-    const haystack = vimState.document.getText();
-    const startOffset = vimState.document.offsetAt(fromPosition);
+    const visualSearch = this.inselection && vimState.lastVisualSelection;
+    const haystack = vimState.document.getText(
```

Comment:
Can you write this like so:
```typescript
let haystack: string;
let searchOffset: number;
if (this.inSelection && vimState.lastVisualSelection) {
  ...
} else {
  ...
}
```

This should make the type assertions unnecessary.

Reply:
👍 will do
I thought you would prefer the constness

Reply:
Yeah this is an annoyance I have with typescript (and C++, which I write in my day job) - rust does it right IMO by making blocks expressions... In any case, I like const-ness, but I care more about type safety.

Reply:
fixed